src/queuemailerl_listener.erl

```diff
@@ -1,71 +1,43 @@
-%% @doc Receives messages from RabbitMQ and dispatches mail sending processes.
+%% @doc A gen_server that listens for events from a RabbitMQ-queue.
+%% It connects to RabbitMQ asynchronusly and starts to subscribe for mail events from
+%% a specified queue. If something goes wrong then this process will tear down everything
+%% and start from scratch.
 -module(queuemailerl_listener).
 
 -export([start_link/0]).
 
 -behaviour(gen_server).
--export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2, code_change/3]).
 
-%% Maximum time to wait when subscribing
--define(SUBSCRIBE_TIMEOUT, 10000).
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2, code_change/3]).
 
 -include_lib("amqp_client/include/amqp_client.hrl").
 
+-define(TIMEOUT, 10000).
+
 %% --- Public API ---
 
 -spec start_link() -> {ok, pid()} | ignore | {error, term()}.
 start_link() ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
 
-%% +--- Gen_server callbacks -----------------------------+
-%% |                                                      |
-%% |  The state of the gen_server is a channel pid only.  |
-%% |                                                      |
-%% +------------------------------------------------------+
+%% --- Gen_server callbacks ---
```

Comment:
What is the state now? If it is not a record called #state{}, it could make sense to comment it. AFAIK it is {ConnectionPid, ChannelPid} | undefined.